internal/provider/catalog/media_datasource.go

```diff
@@ -0,0 +1,171 @@
+				Computed:            true,
+				MarkdownDescription: "True if this media file is in a published catalog.",
+			},
+			"created_at": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "Creation date.",
+			},
+			"size": schema.Int64Attribute{
+				Computed:            true,
+				MarkdownDescription: "Media size in bytes.",
+			},
+			"status": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "Media status.",
+			},
+			"storage_profile": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "Name of the storage profile.",
+			},
+		},
+	}
+}
+
+func (d *catalogMediaDataSource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*client.CloudAvenue)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected *client.CloudAvenue, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	d.client = client
+}
+
+func (d *catalogMediaDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var data catalogMediaDataSourceModel
+	// Read Terraform configuration data into the model
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	catalog, err := d.client.Vmware.Client.GetCatalogById(data.CatalogID.ValueString())
+	if err != nil {
+		tflog.Debug(ctx, "[DEBUG] Unable to find catalog.")
+		resp.Diagnostics.AddError("Unable to find catalog", err.Error())
+		return
+	}
+
+	var media *govcd.Media
+	identifier := data.Name.ValueString()
+	media, err = catalog.GetMediaByNameOrId(identifier, false)
+	if err != nil {
+		tflog.Debug(ctx, "[DEBUG] Unable to find media.")
+		resp.Diagnostics.AddError("Unable to find media", err.Error())
+		return
+	}
+
+	data.ID = types.StringValue(media.Media.ID)
+
+	mediaRecord, err := catalog.QueryMedia(media.Media.Name)
+	if err != nil {
+		tflog.Debug(ctx, "[DEBUG] Unable to query media.")
+		resp.Diagnostics.AddError("Unable to query media", err.Error())
+		return
+	}
+
+	data.CatalogID = types.StringValue(catalog.Catalog.ID)
+	data.Name = types.StringValue(media.Media.Name)
+	data.Description = types.StringValue(media.Media.Description)
+	data.IsISO = types.BoolValue(mediaRecord.MediaRecord.IsIso)
+	data.OwnerName = types.StringValue(mediaRecord.MediaRecord.OwnerName)
+	data.IsPublished = types.BoolValue(mediaRecord.MediaRecord.IsPublished)
+	data.CreatedAt = types.StringValue(mediaRecord.MediaRecord.CreationDate)
+	data.Size = types.Int64Value(mediaRecord.MediaRecord.StorageB)
+	data.Status = types.StringValue(mediaRecord.MediaRecord.Status)
+	data.StorageProfile = types.StringValue(mediaRecord.MediaRecord.StorageProfileName)
+
```

Comment:
For prevent `nil` pointer add check if `catalog.Catalog`, `media.Media` and `media.MediaRecord` is not `nil` before to acceding to values